tests/components/owntracks/test_device_tracker.py

```diff
@@ -1570,7 +1570,6 @@ async def test_restore_state(
     assert state_1 is not state_2
 
     assert state_1.state == state_2.state
-    assert state_1.name == state_2.name
```

Comment:
Why is this check removed from the test?

Reply:
Oh, PR should've been put to draft. But for some reason the name of both devices are different, but I can't seem to figure out why they are different, as they are created from the same data source

Reply:
OK, so the problem is that `device_info` is accessed before `async_added_to_hass` has been called, which means the device name is reset to the config entry name when the entity is restored.

The bug is already there AFAICT, this just PR makes it evident.

I think the solution is to conditionally add the name to the `DeviceInfo` here:
```
    @property
    def device_info(self) -> DeviceInfo:
        """Return the device info."""
        device_info = DeviceInfo(identifiers={(OT_DOMAIN, self._dev_id)})
        if "host_name" in self._data:
            device_info["name"] = self._data["host_name"]
        return device_info
```